examples/pubsub_realtime/nodeset/pubsub_nodeset_rt_publisher.c

```diff
@@ -0,0 +1,681 @@
+/* This work is licensed under a Creative Commons CCZero 1.0 Universal License.
+ * See http://creativecommons.org/publicdomain/zero/1.0/ for more information. */
+
+#define _GNU_SOURCE
+
+/* For thread operations */
+#include <pthread.h>
```

Comment:
I like this extended example. Since the subscriber and publisher tutorials are quite large and complicated, it would be nice to add a comment block regarding the general purpose, functions, parameters and used files of this examples.